shared/constants/types/fs.js

```diff
@@ -48,9 +48,8 @@ export const stringToPath = (s: string): Path => (s.indexOf('/') === 0 ? s : nul
 export const pathToString = (p: Path): string => (!p ? '' : p)
 export const getPathName = (p: Path): string => (!p ? '' : p.split('/').pop())
 export const getPathElements = (p: Path): Array<string> => (!p ? [] : p.split('/').slice(1))
-export const getPathVisibility = (p: Path): Visibility => {
-  if (!p) return null
-  const [, , visibility] = p.split('/')
+export const getVisibilityFromElems = (elems: Array<string>) => {
+  const [, visibility] = elems
```

Comment:
:sheep: `const visibility = elems[1]` is much faster than this

Reply:
True but that also fails if `elems.length < 2`

Reply:
what does that mean?

Reply:
It means that using `const visibility = elems[1]` precipitates the need for an `elems.length` check first. It's fine, I can do it, but I originally used this syntax based on feedback.

Reply:
still confused.
```
const [, v] = [1]
```
turns into
```
var _ref = [1],
    v = _ref[1];
```

you don't need to check any lenght. its value is just undefined